components/molecule/drawer/src/Drawer.js

```diff
@@ -0,0 +1,90 @@
+import {forwardRef, useEffect, useState} from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import useEventListener from '@s-ui/react-hooks/lib/useEventListener'
+import useControlledState from '@s-ui/react-hooks/lib/useControlledState'
+import {ANIMATION_DURATION, PLACEMENTS, SIZES} from './settings'
+
+const MoleculeDrawer = forwardRef(
+  (
+    {
+      animationDuration = ANIMATION_DURATION.FAST,
+      children,
+      isOpen = false,
+      onOpen,
+      onClose,
+      placement = PLACEMENTS.LEFT,
+      size = SIZES.AUTO,
+      target
+    },
+    forwardedRef
+  ) => {
+    const [isOpenState, setIsOpenState] = useControlledState(isOpen) // inner state
+    const [isOpenedState, setIsOpenedState] = useState(isOpen) // transition delayed state
+    useEffect(() => {
+      if (target !== undefined) {
+        target.current.style.position = 'relative'
+        target.current.style.overflow = 'hidden'
+      }
+    }, [target])
+
+    useEventListener('keydown', event => {
+      if (isOpenState === false) return
+      event.preventDefault()
+      if (event.key === 'Escape') {
+        setIsOpenState(false, true)
+      }
+    })
+
+    return (
+      <div
+        onTransitionEnd={event => {
```

Comment:
Could you extract this in a function and then pass it to prop? It's just to keep the return statement cleaner

Reply:
also done